### DISCUSSION TITLE
Exception thrown when filtering collection that contains null values

### DISCUSSION THREAD

**Discussion by austin-best** [alirezanet/Gridify#100] [Ideas]
### Version

2.8.3

### Details

Trying to filter on a collection of data where the property that I'm filtering on might contain null values.
A NullReferenceException is thrown if any of the values that I'm filtering on are null.

### Steps to reproduce

Minimal reproducible example:
```csharp

public class MyThing
{
	public string Name {get;set;}
	public string Mail {get;set;}
}

var myThings = new List<MyThing>()
{
        new MyThing() { Name = "Alpha", Mail = "a@a.com"},
        new MyThing() { Name = "Bravo", Mail = "b@b.com"},
        new MyThing() { Name = "Charlie", Mail = null},
}.AsQueryable();

var queryBuilder = new QueryBuilder<MyThing>();
queryBuilder.AddCondition("Mail$b.com/i");
var results = queryBuilder.Build(myThings).ToList(); // throws Null Reference Exception on materialization
```

Filters that throw an exception:
- `Mail$b@b.com`
- `Mail=b@b.com/i` *(notice, this is case insensitive)*
- `Mail=*com` 

Filters that **do not** throw an exception:
- `Mail=`
- `Mail!=`
- `Mail=b@b.com`  *(notice, this is case sensitive)*

**Comment by alirezanet** [comment]
Hi @austin-best,

What is your expected behavior? asking because you can add a null check if you want, for example these filters works correctly:
```
mail!=,mail$b.com/i
mail!=,mail$a.com
mail=a@a.com
mail!=,mail=*com
```
 `mail!=` you add this part if you want to also add a null check to your expression, however I will still look into this since personally I don't like to see null reference exceptions anywhere 😊.

I had an idea before to add a null safe search settings that always checks for null values before creating actual expressions but since so far no one asked for it It doesn't exist. please let me know what you think

**Comment by austin-best** [comment]
Hi, @alirezanet - 
The null check works, and I'm able to work that into my logic.

I would have expected nulls to have been handled as falsey for any operator except `mail!=`. Now that I know, I can work around it with the null checks in the filters, but the opt-in to change the behavior would be appreciated.

**Reply by alirezanet** [reply]
I agree, would be a nice additional feature... I'll add it probably in the future versions. 
thanks for the feedback

**Comment by alirezanet** [comment]
Hi @austin-best, 
I was looking at my old issues and discussions and stumbled upon this, I think in the latest versions we have this feature and I forgot to update the discussion, If you're still using Gridify please check out the new [`AvoidNullReference`](https://alirezanet.github.io/Gridify/guide/gridifyGlobalConfiguration#avoidnullreference) option.
